components/platform/include/components/platform/dl.hpp

```diff
@@ -0,0 +1,78 @@
+/* Copyright (c) 2022, Arm Limited and Contributors
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 the "License";
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <string>
+
+#define EXPORT
+#define IMPORT
+
+#if defined(_WIN32)
+#	undef EXPORT
+#	undef IMPORT
+#	define EXPORT __declspec(dllexport)
+#	define IMPORT __declspec(dllimport)
+#endif
+
+#define EXPORT_CLIB extern "C" EXPORT
+#define IMPORT_CLIB extern "C" IMPORT
+
+namespace components
+{
+namespace dl
+{
+/**
+ * @brief Convert a library name to an OS specific library name
+ * 
+ * @param name library name
+ * @return std::string OS library name
+ */
+std::string os_library_name(const std::string &name);
+
+/**
+ * @brief Open a dynamic library
+ * 
+ * @param library_path library path
+ * @return void* library handle
+ */
+void *open_library(const char *library_path);
```

Comment:
Mixing `std::string` and `char *` in one interface should be done with a good reason.
I think, you should consistently use the one or the other type here.

Reply:
replaced with `std::string_view` as these functions do not need to own the underlying strings memory